src/dvr/dvr_rec.c

```diff
@@ -124,8 +128,21 @@ dvr_rec_subscribe(dvr_entry_t *de)
     }
   }
 
-  if(stat(de->de_config->dvr_storage, &st) || !S_ISDIR(st.st_mode)) {
-    tvherror(LS_DVR, "the directory '%s' is not accessible", de->de_config->dvr_storage);
+  errSaved = 0;
+  statRet = stat(de->de_config->dvr_storage, &st);
+  errSaved = errno;
+
+  //If the stat() failed, show the ERRNO.
+  if(statRet != 0) {
+    strerror_r(errSaved, errText, sizeof(errText));
+    tvherror(LS_DVR, "Directory '%s' not accessible.  ERRNO: %d, '%s'", de->de_config->dvr_storage, errSaved, errText);
```

Comment:
I'd do something like `"Directory '%s' not accessible (error %d): %s"`. Although there is no need to print out the error number if it has been translated, so a simple `"Directory '%s' not accessible: %s"` would be sufficient, which would then lead to a log message like `...not accessible: No such file or directory`.

Reply:
This is a debugging message in a log file and not intended for general consumption.

The ERRNO is for technical folk to know exactly which error occurred and the text description is to give the user a better idea of what is going on and where to look should they be so inclined.  Both useful pieces of information.

Reply:
The errno contains exactly the same information as the error string, there is no additional information in it. To make things worse: The errno is platform/architecture dependent, so you need to know which architecture the issue appeared on.

Usually what is done is using `perror` when no "special" logging functions are used, which will lead to the error message (without the errno)  being printed to stderr. There's no need for the actual error code, so it isn't shown.

If the error code would contain additional information that would be a different story.

Reply:
I will use your format.